pageserver/src/tenant/timeline.rs

```diff
@@ -3093,15 +3095,18 @@ impl Timeline {
             Some(self.write_lock.lock().await)
         };
         let mut guard = self.layers.write().await;
+        let to_lsn = self.get_last_record_lsn();
         guard
-            .try_freeze_in_memory_layer(self.get_last_record_lsn(), &self.last_freeze_at)
+            .try_freeze_in_memory_layer(to_lsn, &self.last_freeze_at)
```

Comment:
No special reason, but there are a bunch of call paths that pass LSNs down explicitly, so I've kept the general structure the same to avoid having to reason about whether any of them expects something other than the latest value of get_last_record_lsn()